Interface/AddOns/RayUI/libs/xCombatParser-1.0/eventparameters.md
# Parsed Event Args --- Parameters

### Base Parameters
| `args.`*index* | Description |
| ---:|:--- |
| `timestamp` | Uses the function [time()](http://wow.gamepedia.com/API_time) to reference the time of the event when the client received it from the server. |
| `event` | The full name of the event. This is only useful when looking for a specific event or determining if a special event was triggered. |
| `hideCaster` | This parameter lets you know when there is no adequate source name for the event. The parameter `sourceName` will generally be a nil value. <br><br> __NOTE:__ This parser will recolor Environmental events with relevant names and spell IDs. So if this is set, you **_still_** may want to show the `sourceName`. I am considering turning this off for environmental events. |
| `sourceGUID` | The [GUID](http://wow.gamepedia.com/GUID) of the source.
| `sourceName` | The localized name of the source. If `hideCaster` is set, this value may be `nil`, unless the event is environmental. |
| `sourceFlags` | The [unit flags](http://wow.gamepedia.com/UnitFlag) of the source. See [Combat Object Methods](#NEED-LINK) for a set of parsing functions. May be `nil` if you try parsing it yourself. |
| `sourceRaidFlags` | The [raid flags](http://wow.gamepedia.com/RaidFlag) of the source. See [Raid Target Methods](#NEED-LINK) for a set of parsing functions. |
| `destGUID` | The [GUID](http://wow.gamepedia.com/GUID) of the destination.
| `destName` | The localized name of the destination. |
| `destFlags` | The [unit flags](http://wow.gamepedia.com/UnitFlag) of the destination. See [Combat Object Methods](#NEED-LINK) for a set of parsing functions. |
| `destRaidFlags` | The [raid flags](http://wow.gamepedia.com/RaidFlag) of the destination. See [Raid Target Methods](#NEED-LINK) for a set of parsing functions. |


### Event Prefixes

Events with the following prefixes will also have some additional parameters

| `args.prefix` | Parameters | | |
|:--- |:---:|:---:|:---:|
| `"SWING"` | *none* | | |
| `"RANGE"` | `spellId` | `spellName` | [`spellSchool`](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) |
| `"SPELL"` | `spellId` | `spellName` | [`spellSchool`](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) |
| `"SPELL_PERIODIC"` | `spellId` | `spellName` | [`spellSchool`](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) |
| `"SPELL_BUILDING"` | `spellId` | `spellName` | [`spellSchool`](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) |
| `"ENVIRONMENTAL"` | [`environmentalType`](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Environmental_Type) | | |

> __NOTE:__ The use of `spellSchool` above is specific to the spell that was cast. For example, if you cast a _Holy_ spell that did Arcane + Holy (_Divine_) damage, then `spellSchool` would be _Holy_ and `school` would be _Divine_.


### Event Suffixes
Use `args.suffix == "_????"` to view which additional parameters each suffix has.


#### `_DAMAGE`--- Additional Parameters

This is arguably one of the most complicated suffixes and will have its own section dealing with it.

| `args.`*index* | Description |
| ---:|:--- |
| `amount` | The _final_ amount of damage that the destination took from the source in the event. |
| `overkill` | If the destination was killed by the source, this will specify the negative health of the destination. This will be `nil` most of the time. |
| `school` | Specifies the [spell school](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) that was used to damage the destination. It is most useful for coloring the damage amount. |
| `resisted` | Shows how much damage was subtracted from `amount` that was resisted by the destination. You can use `_G["RESIST"]` for a localized string. |
| `blocked` | Shows how much damage was subtracted from `amount` that was blocked by the destination. You can use `_G["BLOCK"]` for a localized string. |
| `absorbed` | Shows how much damage was subtracted from `amount` that was absorbed by the destination. You can use `_G["ABSORB"]` for a localized string. |
| `critical` | A true/false flag that informs if the event was a critical strike. |
| `glancing` | A true/false flag that informs if the event was a glancing blow. |
| `crushing` | A true/false flag that informs if the event was a crushing blow. |
| `isOffHand` | A true/false flag that informs if the event was from the source's off-hand. |


#### `_MISSED`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `missType` | The [type of miss](http://wow.gamepedia.com/COMBAT_LOG_EVENT#Miss_type) that occured. |
| `isOffHand` | A true/false flag that informs if the event was from the source's off-hand. |
| `amountMissed` | The potential damage that no one will ever see or care about. |


#### `_HEAL`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `amount` | The amount of healing that the destination unit from the source. |
| `overhealing` | The amount of healing that was over the maximum health of the destination unit. |
| `absorbed` | The amount of healing that was absorbed by the unit (e.g. by a debuff). |
| `critical` | A true/false flag that informs if the event was a critical strike. |


#### `_ENERGIZE`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `amount` | The amount of energy that the destination unit gained. |
| `powerType` | The [type of power](#power-type-lookup) that the destination unit gained. This used to be an index, but now is just a string with the type. See the table below for the possible values.<br><br>__NOTE:__ If you need a localized string of the power type, you can do:<br><br>`local name = _G[args.powerType];`<br><br>If you need the color of the power type, Blizzard provides that in their `PowerBarColor` global:<br><br>`local color = PowerBarColor[args.powerType]`<br>`local red, green, blue = color.r, color.g, color.b;`|

> #### Power Type Lookup
> | Index (Old) | Power Type | Color (RGB) | Color (Hex) |
> | :---: | :--- | :--- | :--- |
> | 0 | `"MANA"` |  rgb(0, 0, 1) | 0000FF |
> | 1 | `"RAGE"` | rgb(1, 0, 0) | FF0000 |
> | 2 | `"FOCUS"` | rgb(1, 0.5, 0.25) | FF8040 |
> | 3 | `"ENERGY"` | rgb(1, 1, 0) | FFFF00 |
> | 4 | `"CHI"` | rgb(0.71, 1, 0.92) | B5FFEB |
> | 5 | `"RUNES"` | rgb(0.5, 0.5, 0.5) | 808080 |
> | 6 | `"RUNIC_POWER"` | rgb(0, 0.82, 1) | 00D1FF |
> | 7 | `"SOUL_SHARDS"` | rgb(0.5, 0.32, 0.55) | 80528C |
> | 8 | `"LUNAR_POWER"` | rgb(0.3, 0.52, 0.9) | 4D85E6 |
> | 9 | `"HOLY_POWER"` | rgb(0.95, 0.9, 0.6) | F2E699 |
> | 11 | `"MAELSTROM"` | rgb(0, 0.5, 1) | 0080FF |
> | 13 | `"INSANITY"` | rgb(0.4, 0, 0.8) | 6600CC|
> | 17 | `"FURY"` | rgb(0.788, 0.259, 0.992) | C942FD |
> | 18 | `"PAIN"` | rgb(1, 0.612, 0) | FF9C00 |
> | n/a | `"AMMOSLOT"` | rgb(0.8, 0.6, 0) | CC9900  |
> | n/a | `"ARCANE_CHARGES"` | rgb(0.1, 0.1, 0.98) | 1A1AFA |
> | n/a | `"COMBO_POINTS"` | rgb(1, 0.96, 0.41) | FFF569 |
> | n/a | `"FUEL"` | rgb(0, 0.55, 0.5) | 008C80 |
> | n/a | `"STAGGER"` | [1] = rgb(0.52, 1, 0.52)<br>[2] = rgb(1, 0.98, 0.72)<br>[3] = rgb(1, 0.42, 0.42) |  85FF85 _-- Light_ <br>FFFAB8 _-- Moderate_<br>FF6B6B _-- Heavy_ |


#### `_DRAIN` and `_LEECH`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `amount` | The amount that was drained/leeched from the destination to the source. |
| `powerType` | The [type of power](#power-type-lookup) that was drained/leeched.<br><br>__NOTE:__ I believe if this is -2, health was drained/leeched. Also, the new minor stat **Leech** does not seem to use this event. Rather you will find it with `SPELL_HEAL` and `args.spellId == 143924`. |
| `extraAmount` | The amount that was over the power type maximum of the source unit. |


#### `_INTERRUPT`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `extraSpellId` | The spell ID of the spell that was interrupted.<br><br>__NOTE:__ You can use `args.spellId`, `args.spellName`, and `args.spellSchool` to see relevant information about the spell that did the interrupt. |
| `extraSpellName` | The localized name of the spell that was interrupted. |
| `extraSchool` | The school of the spell that was interrupted. |


#### `_DISPEL` and `_STOLEN`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `extraSpellId` | The spell ID of the aura that was dispelled/stolen.<br><br>__NOTE:__ You can use `args.spellId`, `args.spellName`, and `args.spellSchool` to see relevant information about the spell that did the dispelling/stealing. |
| `extraSpellName` | The localized name of the aura that was dispelled/stolen. |
| `extraSchool` | The school of the aura that was dispelled/stolen. |
| `auraType` | The type of aura that was dispelled/stolen: `"BUFF"` or `"DEBUFF"`. |


#### `_DISPEL_FAILED`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `extraSpellId` | The spell ID of the aura that wasn't dispelled.<br><br>__NOTE:__ You can use `args.spellId`, `args.spellName`, and `args.spellSchool` to see relevant information about the spell that tried to dispel. |
| `extraSpellName` | The localized name of the aura that wasn't dispelled. |
| `extraSchool` | The school of the aura that wasn't dispelled. |

#### `_EXTRA_ATTACKS`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `amount` | _Unknown value. More research or documentation is needed._ |


#### `_AURA_APPLIED`, `_AURA_REMOVED`, `_AURA_APPLIED_DOSE	`,  `_AURA_REMOVED_DOSE`, and `_AURA_REFRESH`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `auraType` | The type of the aura being referenced in the event (`"BUFF"` or `"DEBUFF"`).  |
| `amount` | The number of stacks of the aura. |


#### `_AURA_BROKEN`--- Additional Parameters
| `args.`*index* | Description |
| ---:|:--- |
| `auraType` | The type of the aura being broken in the event (`"BUFF"` or `"DEBUFF"`).  |


#### `_AURA_BROKEN_SPELL`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `extraSpellId` | The spell ID of the aura that was broken.<br><br>__NOTE:__ You can use `args.spellId`, `args.spellName`, and `args.spellSchool` to see relevant information about the spell that did the breaking. |
| `extraSpellName` | The localized name of the aura that was broken. |
| `extraSchool` | The school of the aura that was broken. |
| `auraType` | The type of aura that was broken: `"BUFF"` or `"DEBUFF"`. |


#### `_CAST_FAILED`--- Additional Parameters

| `args.`*index* | Description |
| ---:|:--- |
| `failedType` | A localized string that gives the reason the spell failed.<br><br>__Examples:__<br><br>`"Can only use outside"`<br>`"Item is not ready yet"`<br>`"No target"`<br>...<br>_etc._|


> Written with [StackEdit](https://stackedit.io/).


Interface/AddOns/Skada/Options.lua
local L = LibStub("AceLocale-3.0"):GetLocale("Skada", false)
local media = LibStub("LibSharedMedia-3.0")

Skada.resetoptions = {[1] = L["No"], [2] = L["Yes"], [3] = L["Ask"]}

Skada.windowdefaults = {
	name = "Skada",

	barspacing=0,
	bartexture="BantoBar",
	barfont="Accidental Presidency",
	barfontflags="",
	barfontsize=13,
	barheight=18,
	barwidth=240,
	barorientation=1,
	barcolor = {r = 0.3, g = 0.3, b = 0.8, a=1},
	barbgcolor = {r = 0.3, g = 0.3, b = 0.3, a = 0.6},
	barslocked=false,
	clickthrough=false,

	classcolorbars = true,
	classcolortext = false,
	classicons = true,
	roleicons = false,
    showself = true,

	buttons = {menu = true, reset = true, report = true, mode = true, segment = true},

	title = {textcolor = {r = 0.9, g = 0.9, b = 0.9, a = 1}, height = 20, font="Accidental Presidency", fontsize=13, texture="Armory", bordercolor = {r=0,g=0,b=0,a=1}, bordertexture="None", borderthickness=2, color = {r=0.3,g=0.3,b=0.3,a=1}, fontflags = ""},
	background = {
        height=200,
        texture="Solid",
        bordercolor = {r=0,g=0,b=0,a=1}, 
        bordertexture="Blizzard Party", 
        borderthickness=2, 
        color = {r=0,g=0,b=0,a=0.4}, 
        tile = false, 
        tilesize = 0, 
    },
    
    strata = "LOW",
    scale = 1,

	reversegrowth=false,
	modeincombat="",
	returnaftercombat=false,
	wipemode="",

	hidden = false,
	enabletitle = true,
	titleset = true,

	set = "current",
	mode = nil,

	display = "bar",
	snapto = true,
	scale = 1,
    version = 1,
    
    -- Inline exclusive
    isonnewline = false,
    isusingclasscolors = true,
    height = 30,
    width = 600,
    color = {r = 0.3, g = 0.3, b = 0.3, a = 0.6},
    isusingelvuiskin = true,
    issolidbackdrop = false,
    fixedbarwidth = false,
    
    -- Broker exclusive
    textcolor = {r = 0.9, g = 0.9, b = 0.9},
    useframe = true
}

local windefaultscopy = {}
Skada:tcopy(windefaultscopy, Skada.windowdefaults)

Skada.defaults = {
	profile = {
		version=1,
		reset={instance=1, join=3, leave=1},
		icon = {hide = false, radius = 80, minimapPos = 195},
		numberformat=1,
		setformat=3,
		setnumber=true,
		showranks=true,
		setstokeep=10,
		tooltips=true,
		tooltippos="smart",
		tooltiprows=3,
		informativetooltips=true,
		onlykeepbosses=false,
		tentativecombatstart=false,
		hidesolo=false,
		hidepvp=false,
		hidedisables=true,
		hidecombat=false,
		mergepets=true,
		feed = "",
		showtotals = false,
        autostop = false,

		modules = {},
		columns = {},
		report = {mode = "Damage", set = "current", channel = "Say", chantype = "preset", number = 10},
		modulesBlocked = {
		},

		windows = {windefaultscopy},
	}
}

-- Adds column configuration options for a mode.
function Skada:AddColumnOptions(mod)
	local db = self.db.profile.columns

	if mod.metadata and mod.metadata.columns then
		local cols = {
				type = "group",
				name = mod:GetName(),
				order=0,
				inline=true,
				args = {}
		}

		for colname, value in pairs(mod.metadata.columns) do
			local c = mod:GetName().."_"..colname

			-- Set initial value from db if available, otherwise use mod default value.
			if db[c] ~= nil then
				mod.metadata.columns[colname] = db[c]
			end

			-- Add column option.
			local col = {
					type="toggle",
					name=L[colname] or colname,
					get=function() return mod.metadata.columns[colname] end,
					set=function()
								mod.metadata.columns[colname] = not mod.metadata.columns[colname]
								db[c] = mod.metadata.columns[colname]
								Skada:UpdateDisplay(true)
							end,
			}
			cols.args[c] = col
		end

		Skada.options.args.columns.args[mod:GetName()] = cols
	end
end

function Skada:AddLoadableModuleCheckbox(mod, name, description)
	local new = {
		type = "toggle",
		name = name,
        desc=description,
		order=1,
	}
	Skada.options.args.modules.args[mod] = new
end

local deletewindow = nil
local newdisplay = "bar"
Skada.options = {
	type="group",
	name="Skada",
	plugins = {},
	args={
			d = {
				type="description",
				name=L["A damage meter."],
				order=0,
			},

			windows = {
				type = "group",
				name = L["Windows"],
				order=1,
				args = {

					create = {
						type="input",
						name=L["Create window"],
						desc=L["Enter the name for the new window."],
						set=function(self, val) if val and val ~= "" then Skada:CreateWindow(val, nil, newdisplay) end end,
						order=1,
					},

					display = {
						type="select",
						name=L["Display system"],
						desc=L["Choose the system to be used for displaying data in this window."],
						values=	function()
									local list = {}
									for name, display in pairs(Skada.displays) do
										list[name] = display.name
									end
									return list
								end,
						get=function() return newdisplay end,
						set=function(i, display) newdisplay = display end,
						order=2,
					},

				},
			},

			resetoptions = {
				type = "group",
				name = L["Data resets"],
				order=2,
				args = {

					resetinstance = {
						type="select",
						name=L["Reset on entering instance"],
						desc=L["Controls if data is reset when you enter an instance."],
						values=	function() return Skada.resetoptions end,
						get=function() return Skada.db.profile.reset.instance end,
						set=function(self, opt) Skada.db.profile.reset.instance = opt end,
						order=30,
					},

					resetjoin = {
						type="select",
						name=L["Reset on joining a group"],
						desc=L["Controls if data is reset when you join a group."],
						values=	function() return Skada.resetoptions end,
						get=function() return Skada.db.profile.reset.join end,
						set=function(self, opt) Skada.db.profile.reset.join = opt end,
						order=31,
					},

					resetleave = {
						type="select",
						name=L["Reset on leaving a group"],
						desc=L["Controls if data is reset when you leave a group."],
						values=	function() return Skada.resetoptions end,
						get=function() return Skada.db.profile.reset.leave end,
						set=function(self, opt) Skada.db.profile.reset.leave = opt end,
						order=32,
					},

				}

			},

			tooltips = {
				type = "group",
				name = L["Tooltips"],
				order=3,
				args = {
					tooltips = {
							type="toggle",
							name=L["Show tooltips"],
							desc=L["Shows tooltips with extra information in some modes."],
							order=1,
							get=function() return Skada.db.profile.tooltips end,
							set=function() Skada.db.profile.tooltips = not Skada.db.profile.tooltips end,
					},

					informative = {
							type="toggle",
							name=L["Informative tooltips"],
							desc=L["Shows subview summaries in the tooltips."],
							order=2,
							get=function() return Skada.db.profile.informativetooltips end,
							set=function() Skada.db.profile.informativetooltips = not Skada.db.profile.informativetooltips end,
					},

					rows = {
						type="range",
						name=L["Subview rows"],
						desc=L["The number of rows from each subview to show when using informative tooltips."],
						min=1,
						max=10,
						step=1,
						get=function() return Skada.db.profile.tooltiprows end,
						set=function(self, val) Skada.db.profile.tooltiprows = val end,
						order=3,
					},

					tooltippos = {
						type="select",
						name=L["Tooltip position"],
						desc=L["Position of the tooltips."],
						values=	{["default"] = L["Default"], ["topright"] = L["Top right"], ["topleft"] = L["Top left"], ["smart"] = L["Smart"]},
						get=function() return Skada.db.profile.tooltippos end,
						set=function(self, opt) Skada.db.profile.tooltippos = opt end,
						order=4,
					},
				}
			},

			generaloptions = {
				type = "group",
				name = L["General options"],
				order=4,
				args = {

					mmbutton = {
							type="toggle",
							name=L["Show minimap button"],
							desc=L["Toggles showing the minimap button."],
							order=1,
							get=function() return not Skada.db.profile.icon.hide end,
							set=function()
									Skada.db.profile.icon.hide = not Skada.db.profile.icon.hide
									Skada:RefreshMMButton()
								end,
					},

					mergepets = {
							type="toggle",
							name=L["Merge pets"],
							desc=L["Merges pets with their owners. Changing this only affects new data."],
							order=2,
							get=function() return Skada.db.profile.mergepets end,
							set=function() Skada.db.profile.mergepets = not Skada.db.profile.mergepets end,
					},

					showtotals = {
							type="toggle",
							name=L["Show totals"],
							desc=L["Shows a extra row with a summary in certain modes."],
							order=3,
							get=function() return Skada.db.profile.showtotals end,
							set=function() Skada.db.profile.showtotals = not Skada.db.profile.showtotals end,
					},

					onlykeepbosses = {
							type="toggle",
							name=L["Only keep boss fighs"],
							desc=L["Boss fights will be kept with this on, and non-boss fights are discarded."],
							order=4,
							get=function() return Skada.db.profile.onlykeepbosses end,
							set=function() Skada.db.profile.onlykeepbosses = not Skada.db.profile.onlykeepbosses end,
					},

					hidesolo = {
							type="toggle",
							name=L["Hide when solo"],
							desc=L["Hides Skada's window when not in a party or raid."],
							order=5,
							get=function() return Skada.db.profile.hidesolo end,
							set=function()
										Skada.db.profile.hidesolo = not Skada.db.profile.hidesolo
										Skada:ApplySettings()
									end,
					},

					hidepvp = {
							type="toggle",
							name=L["Hide in PvP"],
							desc=L["Hides Skada's window when in Battlegrounds/Arenas."],
							order=6,
							get=function() return Skada.db.profile.hidepvp end,
							set=function()
										Skada.db.profile.hidepvp = not Skada.db.profile.hidepvp
										Skada:ApplySettings()
									end,
					},

					hidecombat = {
							type="toggle",
							name=L["Hide in combat"],
							desc=L["Hides Skada's window when in combat."],
							order=7,
							get=function() return Skada.db.profile.hidecombat end,
							set=function()
										Skada.db.profile.hidecombat = not Skada.db.profile.hidecombat
										Skada:ApplySettings()
									end,
					},

					disablewhenhidden = {
							type="toggle",
							name=L["Disable while hidden"],
							desc=L["Skada will not collect any data when automatically hidden."],
							order=8,
							get=function() return Skada.db.profile.hidedisables end,
							set=function()
										Skada.db.profile.hidedisables = not Skada.db.profile.hidedisables
										Skada:ApplySettings()
									end,
					},

					numberformat = {
						type="select",
						name=L["Number format"],
						desc=L["Controls the way large numbers are displayed."],
						values=	function() return {[1] = L["Condensed"], [2] = L["Detailed"]} end,
						get=function() return Skada.db.profile.numberformat end,
						set=function(self, opt) Skada.db.profile.numberformat = opt end,
						order=11,
					},

					showranks = {
							type="toggle",
							name=L["Show rank numbers"],
							desc=L["Shows numbers for relative ranks for modes where it is applicable."],
							order=9,
							get=function() return Skada.db.profile.showranks end,
							set=function()
										Skada.db.profile.showranks = not Skada.db.profile.showranks
										Skada:ApplySettings()
									end,
					},

					datafeed = {
						type="select",
						name=L["Data feed"],
						desc=L["Choose which data feed to show in the DataBroker view. This requires an LDB display addon, such as Titan Panel."],
						values=	function()
									local feeds = {}
									feeds[""] = L["None"]
									for name, func in pairs(Skada:GetFeeds()) do feeds[name] = name end
									return feeds
								end,
						get=function() return Skada.db.profile.feed end,
						set=function(self, feed)
									Skada.db.profile.feed = feed
									if feed ~= "" then Skada:SetFeed(Skada:GetFeeds()[feed]) end
								end,
						order=12,
					},

					setstokeep = {
						type="range",
						name=L["Data segments to keep"],
						desc=L["The number of fight segments to keep. Persistent segments are not included in this."],
						min=0,
						max=99,
						step=1,
						get=function() return Skada.db.profile.setstokeep end,
						set=function(self, val) Skada.db.profile.setstokeep = val end,
						order=13,
					},

					setnumber = {
							type="toggle",
							name=L["Number set duplicates"],
							desc=L["Append a count to set names with duplicate mob names."],
							order=13.5,
							get=function() return Skada.db.profile.setnumber end,
							set=function() Skada.db.profile.setnumber = not Skada.db.profile.setnumber end,
					},

					setformat = {
						type="select",
						name=L["Set format"],
						desc=L["Controls the way set names are displayed."],
						values=	Skada:SetLabelFormats(),
						get=function() return Skada.db.profile.setformat end,
						set=function(self, opt) Skada.db.profile.setformat = opt; Skada:ApplySettings(); end,
						order=14,
						width="double",
					},

					tenativecombatstart = {
							type="toggle",
							name=L["Aggressive combat detection"],
							desc=L["Skada usually uses a very conservative (simple) combat detection scheme that works best in raids. With this option Skada attempts to emulate other damage meters. Useful for running dungeons. Meaningless on boss encounters."],
							order=10,
							get=function() return Skada.db.profile.tentativecombatstart end,
							set=function() Skada.db.profile.tentativecombatstart = not Skada.db.profile.tentativecombatstart end,
					},
                
					autostop = {
							type="toggle",
							name=L["Autostop"],
							desc=L["Autostop description"],
							order=10,
							get=function() return Skada.db.profile.autostop end,
							set=function() Skada.db.profile.autostop = not Skada.db.profile.autostop end,
					},
                
					showself = {
							type="toggle",
							name=L["Always show self"],
							desc=L["Keeps the player shown last even if there is not enough space."],
							order=11,
							get=function() return Skada.db.profile.showself end,
							set=function() 
                                Skada.db.profile.showself = not Skada.db.profile.showself
                                Skada:ApplySettings()
                            end,
					},
                
                
				}
			},
			columns = {
				type = "group",
				name = L["Columns"],
				order=5,
				args = {},
			},
			modules = {
				type = "group",
				name = L["Disabled Modules"],
				order=6,
				get = function(i) return Skada.db.profile.modulesBlocked[i[#i]] end,
				set = function(i, value) Skada.db.profile.modulesBlocked[i[#i]] = value; Skada.options.args.modules.args.apply.disabled=false end,
				args = {
					desc = {
						type="description",
						name=L["Tick the modules you want to disable."],
						width="full",
						order=0,
					},
					apply = {
						type="execute",
						name=APPLY,
						width="full",
						func=ReloadUI,
						confirm = function()
							return L["This change requires a UI reload. Are you sure?"]
						end,
						disabled = true,
						order=99,
					},
				},
			}
	}
}
